src/edu/cscc/CustomerRepositoryImpl.java

```diff
@@ -0,0 +1,96 @@
+package edu.cscc;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+
+public class CustomerRepositoryImpl implements CustomerRepository {
+
+    private Set<Customer> customers;
+
+    public CustomerRepositoryImpl(){
+        customers = new HashSet<>();
+    }
+
+    public Customer create(Customer customer) throws DuplicateCustomerException {
+        boolean foundMatch = customers.stream()
+                .anyMatch(customer1 -> customer1.getEmailAddress().equals(customer.getEmailAddress()));
+
+        if (foundMatch) {
+            throw new DuplicateCustomerException(customer.getEmailAddress());
+        } else {
+            customer.setAccountNumber(UUID.randomUUID());
+            customers.add(customer);
+            return customer;
+        }
+    }
+
+    public Customer read(UUID accountNumber){
+        Optional<Customer> accountFound = customers.stream()
+                .filter(customer -> customer.getAccountNumber() == accountNumber)
+                .findFirst();
+
+        if (accountFound.isPresent()) {
+            return accountFound.get();
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public Customer findByEmailAddress(String emailAddress) {
+        Optional<Customer> emailFound = customers.stream()
+                .filter(customer -> customer.getEmailAddress().toLowerCase().equals(emailAddress.toLowerCase()))
+                .findFirst();
+
+        if (emailFound.isPresent()) {
+            return emailFound.get();
+        } else {
+            return null;
+        }
+    }
+
+    public void update(Customer customer) throws DuplicateCustomerException, CustomerNotFoundException {
+        Optional<Customer> outdatedCustomer = customers.stream()
+                .filter(customer1 -> customer1.getAccountNumber() == customer.getAccountNumber())
+                .findFirst();
+
+        if (outdatedCustomer.isPresent()){
+            //Found the customer. If the email has changed, make sure it doesn't match any other customer
+            if (outdatedCustomer.get().getEmailAddress().equals(customer.getEmailAddress())){
+                outdatedCustomer.get().setLastName(customer.getLastName());
+                outdatedCustomer.get().setFirstName(customer.getLastName());
```

Comment:
Whoops, it looks like you forgot to set the account number for the customer, which is causing a NullPointerException when we attempt to update the program. 

Steps to recreate:
1. Run the main program.
2. Create a customer.
3. Choose the option to update the customer.
4. Enter the account number of the previously created customer.
5. Observe the NullPointerException and stack trace that is generated.